### ISSUE TITLE
 bug on Line 138 in ical.net.collections.groupedlist.Remove(TItem obj)

### ISSUE THREAD

**Issue by posics** [ical-org/ical.net#143]
if this is a valid vcal

BEGIN:VCALENDAR
PRODID:-//github.com/rianjs/ical.net//NONSGML ical.net 2.2//EN
VERSION:2.0
BEGIN:VEVENT
CATEGORIES:Free slot
DTEND:20160920T170000Z
DTSTAMP:20160920T100557Z
DTSTART:20160920T050000Z
SEQUENCE:0
UID:9a6c0cd1-ee43-47bb-9733-9552e778f602
END:VEVENT
BEGIN:VEVENT
CATEGORIES:Opening times
DTEND:20160920T170000Z
DTSTAMP:20160920T094210Z
DTSTART:20160920T050000Z
EXRULE:FREQ=WEEKLY;BYDAY=SU
RRULE:FREQ=DAILY
SEQUENCE:0
UID:97ac7136-8537-4bf3-aec6-bde82aa52044
END:VEVENT
END:VCALENDAR

then I think I've found a bug on Line 138 in ical.net.collections.groupedlist.Remove(TItem obj)
If, on the other hand, this is an illegal vcal then there is no bug so you don't have to read through just remove the issue.

For some reason or another line 138 returns indexof(obj) as 0 (which is my "Free slot" event) instead of 1 (which are my "Opening times") even though I said cal.Events.Remove(openingTimes); (which is correctly set to and received by Remove(TItem obj) as opening times as can be seen in a debugging session) and UID is different for both;

Basically I have a caledar with Opening times for an office and if there are no events planned for the day then I add a Free slot so that anyone can book a meeting whenever they want, inside of Opening times. The quirk is that both start and end times are the same for the two, timestamp however as well as uid are different.

**Comment by rianjs** [comment]
@posics - Do you have a C# snippet that causes the unexpected behavior?

**Comment by posics** [comment]
Of course, it's fairly mundane bit of code. When the above vcal string is deserialized into a calendar and OfficeHours being the "Opening times" event with UID "97ac7136-8537-4bf3-aec6-bde82aa52044" and calendar.Events sent to this function as sourceEvents:
        public static ICalendar GenerateCalendarWithFreeSlots(IEvent OfficeHours, List<IEvent> sourceEvents)
        {
            var targetCalendar = new Calendar();
            if (sourceEvents == null || sourceEvents.Count == 0) sourceEvents = new List<IEvent>();
            else if (sourceEvents.Count > 0 && OfficeHours != null) sourceEvents.Remove(OfficeHours);
etc etc...
}

Maybe I could have just written the whole deserialize-chopup-function call code instead of just explaining here? My bad...
